src/chonkie/handshakes/weaviate.py

```diff
@@ -209,7 +207,7 @@ def _collection_exists(self, collection_name: str) -> bool:
             exists = self.client.collections.exists(collection_name)
             return exists
         except Exception as e:
-            print(f"Warning: Failed to check for collection '{collection_name}': {e}")
+            logger.warning(f"Failed to check for collection '{collection_name}': {e}")
             return False
```

Comment:
maybe use raise here instead of false, would be a nice patch :) 
```suggestion
            return False
```